app/src/test/java/com/globant/harrypotterapp/characterdetail/CharacterDetailViewModelTest.kt

```diff
@@ -75,28 +77,50 @@ class CharacterDetailViewModelTest {
             characterDetailViewModel.fetchCharacterDetail(HARRY_POTTER_ID).join()
         }
 
+        successResponseList[ONE].data?.let { verify(mockedDatabase).updateCharacterDetail(it) }
         verify(mockedCharacterDetailService).getCharacterDetail(HARRY_POTTER_ID)
 
-        Assert.assertEquals(successResponseList[ZERO].status, charactersLiveData.observedValues[ZERO]?.peekContent()?.status)
-        Assert.assertEquals(successResponseList[ONE].status, charactersLiveData.observedValues[ONE]?.peekContent()?.status)
-        Assert.assertEquals(successResponseList[ONE].data, charactersLiveData.observedValues[ONE]?.peekContent()?.data)
+        assertEquals(successResponseList[ZERO].status, charactersLiveData.observedValues[ZERO]?.peekContent()?.status)
+        assertEquals(successResponseList[ONE].status, charactersLiveData.observedValues[ONE]?.peekContent()?.status)
+        assertEquals(successResponseList[ONE].data, charactersLiveData.observedValues[ONE]?.peekContent()?.data)
     }
 
     @Test
-    fun `when fetchCharacterDetail return failure result`() {
+    fun `when fetchCharacterDetail return failure result, then gets character detail from database`() {
         val charactersLiveData = characterDetailViewModel.getCharacterDetailLiveData().testObserver()
 
-        whenever(mockedCharacterDetailService.getCharacterDetail(WRONG_HARRY_POTTER_ID)).thenReturn(Result.Failure(exception))
+        whenever(mockedCharacterDetailService.getCharacterDetail(HARRY_POTTER_ID)).thenReturn(Result.Failure(exception))
+        whenever(mockedDatabase.getCharacterDetail(HARRY_POTTER_ID)).thenReturn(Result.Success(data = characterDetail))
+
+        runBlocking {
+            characterDetailViewModel.fetchCharacterDetail(HARRY_POTTER_ID).join()
+        }
+
+        verify(mockedCharacterDetailService).getCharacterDetail(HARRY_POTTER_ID)
+        verify(mockedDatabase).getCharacterDetail(HARRY_POTTER_ID)
+
+        assertEquals(successResponseList[ZERO].status, charactersLiveData.observedValues[ZERO]?.peekContent()?.status)
+        assertEquals(successResponseList[ONE].status, charactersLiveData.observedValues[ONE]?.peekContent()?.status)
+        assertEquals(successResponseList[ONE].data, charactersLiveData.observedValues[ONE]?.peekContent()?.data)
```

Comment:
pls can you change, ZERO & ONE for FIRST_RESPONSE and SECOND_RESPONSE?

Reply:
Refactored!